Chapter3/README.md

```diff
@@ -0,0 +1,205 @@
+
+![publish](../Misc/ch3_publish.png)
+
+#### Step 4. Configure your app
+
+On the **App Services** portal, click on the **Configuration** blade and set the **stack** property to **.NET Core**.
+
+![config](../Misc/ch3_config.png)
+
+#### Step 5. Disabling Azure AD authentication
+
+Still on the **App Services** portal, click on the **Authentication/Authorization** blade. There, make sure that the **App Services Authentication** is switched off (and nothing else is checked), as we are using our own authentication logic.  
+
+![auth1](../Misc/ch3_auth.png)
+
+### TodoListSPA
+
+Since TodoListSPA is a single-page application, we will deploy it as a **static website** on [Azure Storage](https://azure.microsoft.com/services/storage/). To do so, we will **build** our Angular sample to create a `dist/<your-project-name>` folder with the compiled resources. Once done, We will upload this `dist/<your-project-name>` folder only (and not the entire `dist/` folder).
+
+> A video tutorial covering this process can be found [here](https://docs.microsoft.com/azure/javascript/tutorial-vscode-static-website-node-01).
+
+Using a command line interface such as VS Code integrated terminal, locate the `TodoListSPA folder`, then follow the steps below:
+
+#### Step 1. Create storage account
+
+On VS Code, click on the Azure Tools icon on the left bar. Then, find the Storage section, and click on one of your subscriptions to create a storage account.
+
+![storage-account](../Misc/ch3_storage_account.png)
+
+#### Step 2. Build your app
+
+```console
+    ng build --prod
+```
+
+#### Step 3. Deploy static website
+
+Right-click on the newly created `dist/<your-project-name>` folder and a context-menu will appear with the option "Deploy to static website".
+
+![deploy-static](../Misc/ch3_deploy_static.png)
+
+You may be prompted you to sign in to Azure, providing a link to open in your browser and a code to paste in the login page. Once deployment is done, a popup on the right corner of VS Code will appear to let you know the URL of the website you've just deployed.
+
+#### Step 4. Register your new address as a redirect uri
+
+You now need to go back to your Azure AD **app registration** for `TodoListSPA`:
+
+   1. Click on the **Authentication** blade.
+   2. Add your website's address (e.g. `https://angular9-todo-app52.z22.web.core.windows.net/`) as a new **Redirect URI**.
+
+#### Step 5. Modify app-config.json
+
+1. Open the `TodoListSPA\src\app\app-config.json` file
+1. Find the key `todoListApi.redirectUri` and replace the existing value with the **Redirect Uri** that you've just registered in the previous step.
+1. Find the key `todoListApi.resourceScope` and replace the existing value with the **Redirect Uri** that you've just obtained in deploying TodoListAPI (e.g. my-web-api.azurewebsites.net)
```

Comment:
```suggestion
1. Find the key `resources.todoListApi.resourceScope` and replace it with the scope value found in the Azure Portal under Azure Active Directory/App Registrations/**TodoListAPI**/Expose API blade (e.g. `api://{{client_id}}/access_as_user`)
1. Find the key `resources.todoListApi.resourceUri` and replace the existing value with the **Redirect Uri** that you've just obtained in deploying **TodoListAPI** (e.g. `my-web-api.azurewebsites.net/api/todolist`)
```

Reply:
actually I made a mistake here. Scope part need not to change, just to resourceUri. fixed now

Reply:
It should say ` (e.g. `my-web-api.azurewebsites.net/api/todolist)` . If the instructions here are followed precisely, the user will be missing the `/api/todolist` and the app won't work.